File: subsets/subsets.cs
public class Solution {
    List<IList<int>> result;
    public IList<IList<int>> Subsets(int[] nums) {
        result = new List<IList<int>>();
        if(nums==null || nums.Length ==0)
        {
            return result;
        }
        
        backtrack(nums,new List<int>(),0);
        return result;
    }
    
    private void backtrack(int[] nums,List<int> temp, int index)
    {
        
    
            result.Add(temp);
           
        
        for(int i=index;i<nums.Length;i++)
        {
            temp.Add(nums[i]);
            backtrack(nums,new List<int>(temp),i+1);
            temp.RemoveAt(temp.Count-1);
        }
    }
}

File: .idea/subsets.java
class subsets {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> result = new ArrayList<>();
        result.add(new ArrayList<>());
        for(int num: nums)
        {
            int size = result.size();
            for(int i=0; i< size; i++)
            {
                List<Integer> temp = new ArrayList<>(result.get(i));
                temp.add(num);
                result.add(temp);
            }
        }
        return result;
    }
}